### DISCUSSION TITLE
[Feedback] New Reanimated Documentation

### DISCUSSION THREAD

**Discussion by kacperkapusciak** [software-mansion-labs/reanimated-beta-docs#12] [Announcements]
This repository is the home of the current developments regarding the Reanimated Beta Docs. This project, when ready, will take the place of [the current stable documentation](https://docs.swmansion.com/react-native-reanimated/).

**The beta version of the documentation is available at [reanimated-beta-docs.swmansion.com](https://reanimated-beta-docs.swmansion.com/).**

A lot of content like some of the hooks, LayoutAnimations and Shared Element Transitions aren't explained yet. We'd like to collect the early feedback to further iterate on the content.

We're excited to hear about your ideas, insights and the overall feeling of the site.

Looking forward to hear from you!

**Comment by IslamRustamov** [comment]
Hello!

Great job on new documentation. What I can pinpoint (in my opinion) is that it got kind of easier to read it, comparing to the old documentation (font size, shorter and simpler explanation, that kind of stuff). 

Realtime examples of animations is a great plus. What do you think about the idea of creating a separate chapter which is dedicated only to examples of different kinds of animations which might be used/seen in real projects? I think it would be easier for developers to just take a look there, find the most suitable animation for their particular case and copy-paste it (though it might be better if developers learnt how to build these animations themselves).

Also about the [cancelAnimation](https://reanimated-beta-docs.swmansion.com/docs/core/cancelAnimation). As much as I understand, to resume animation we need to call *withTiming/withDelay/withSpring/...* again, but maybe some sort of *resumeAnimation* would look better?

Thanks in advance.

**Reply by kacperkapusciak** [reply]
Thanks for feedback! 🙏 

### 1

Amazing feeling to hear this  🙌  

### 2

 A separate section in the documentation with _real use-case_ _ready-to-copy-and-paste_ is a great idea that we'd love to implement that 🥇 That's a **yes** from us!
 
### 3

To resume an animation you just to assign the same values as you set initially and we believe it's for the better. I think `resumeAnimation` would make things harder, not easier.

The current state of an animation is kept in a shared value. When you run `cancelAnimation` it cancels the animation by assigning the same shared value to each other causing the progress of an animation to stop. Then if you just assign an animation function _(withTiming/withSpring/...)_ with the same starting conditions to it, it will "resume" as you would expect.

`resumeAnimation` adds additional complexity. Is the animation currently running? Does the user would have to check for that himself? What should happen when no animation was previously running? Throwing an error? An noop? Far from ideal.

Damn. I have to document that! 🏃

**Comment by jozan** [comment]
i've been using the new docs site for a few days. i'm not a first time user but haven't been using reanimated v2 since it's first release.

1. the front page looks awesome. i was excited to add animations to my app and when i saw the new site i was like a kid in a candy shop. it's been a years since i've used reanimated and i was already sold by then so cannot comment how convincing the front page is for first timers. i just clicked the big blue button and it was shopping time for me. my app needed some useSharedValues and withSprings and the shelves were stocked.

2. the left sidebar is my most favorite thing. having all things there  visible, neatly organized and searchable with ctrl/cmd+f has been valuable. on the old site i don't like that things are hidden and i needed to find them by clicking kinda randomly to reveal more content. i'm used to this in "your face pattern" on many other documentation sites and it's good. i hope you'll keep it.

3. functionality wise it just works as expected. safari on mac and firefox on windows, it looks all the same. search also works. i like the light mode color scheme so i used it. not comment on dark mode, i haven't used it. toggling the mode was easy to find which is great design.

4. i dived straight into api documentation of hooks and layout animation so i haven't really read all those guides and stuff. i'm not dismissing the effort that is put to them. i just haven't needed them yet! i guess it says something about the api design and how good it actually is.

5. interactive demos are nice but i didn't play with them that much. i already have strong mental model how parameters affect the outcome. i saw [some pendulum tweet](https://twitter.com/aleksandracynk/status/1676972530618486784) a while back and i'm hoping it'll be on the docs site one day.

6. porting more api documentation would be nice. i don't like to hop to the old site that much but it was necessary. i needed to read about layout animations. maybe those could be ported over or rewritten. 

cheers
/j

**Reply by kacperkapusciak** [reply]
Thank you for kind words stranger 🙇 

Ad 1 & 3 Amazing to hear! 🙌 🥹 

Ad 2. I'm also a big fan of the current "open" sidebar and I'd prefer to keep it that way as long as possible. As the documentation grows, there are more and more voices that say that too many shown pages would overwhelm the readers. I'm glad I have someone from outside the team to back my thought process, which is exactly the same as yours.

Ad 4. Rarely anyone reads a documentation from start to finish. Duh, no one has time for that. Most people just skim it over to get the job done. In all the sections, besides the _Fundamentals_ which is directed to beginners, we're trying to optimise the content for quick skimming. Buring information in long paragraphs isn't the best experience for anyone, really.

Ad 5. I think we can make that happen 😉 

Ad 6. We're slowly but surely rewriting the content of the old site. The goal is to fully replace the old documentation by the new one page-by-page (or more like concept-by-concept). More content is coming!

Great feedback🏅 
Thanks for your time! 

Cya
